### PR TITLE
feat: validate address book import

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Address book import validation.

## How this PR fixes it

Entries are first validated, displaying an error/disabling the import button if they are invalid. Entries that contain empty fields are ignored. `chainId`s in address books that have whitespace are sanitised.

### Validation
- Import must be `.csv` file
- It cannot be > 1MB in size
- CSV file must contain data
- CSV file must have the correct headers
- Entries must exist
- All addresses must be valid
- All `chainId`s must be valid

## How to test it

Manipulate an address book CSV to flout one of the above validation rules and observe the shown that prevents submission.

## Screenshots
image

### CODE REVIEW

File: src/components/address-book/ImportDialog/validation.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  // No entries
+  if (entries.length === 0) {
+    return 'No entries found in address book'
+  }
+
+  // An entry has empty value
+  if (!hasCompleteAbEntries(entries)) {
+    return 'Address book contains entries with empty fields'
+  }
```
Comment (usame-algan) [inline src/components/address-book/ImportDialog/validation.ts:63]: Do we need this validation? Currently, we just import everything that can be imported and ignore the rest which is a very simple solution.

Comment (iamacook) [reply src/components/address-book/ImportDialog/validation.ts:63]: That's an elegant solution that I didn't think of. I'll filter out the empty fields then.

Comment (iamacook) [reply src/components/address-book/ImportDialog/validation.ts]: Good catch. I've adjusted it to ignore them.

**Review by katspaugh — APPROVED** [review]
Nice work 👍

**Review by usame-algan — APPROVED** [review]

**Pull request merged**